code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -700,23 +702,21 @@ var/global/list/obj/item/device/pda/PDAs = list()
 	set name = "Remove Pen"
 	set src in usr
 
-	var/obj/item/inserted_item = loc
-
 	if(issilicon(usr))
 		return
+	var/obj/item/weapon/pen/O = locate() in src //Needed, otherwise cannot remove default pen in PDA

```

Comment:
no need to, just make sure the pen spawned inside the pda in pda/New() is given to inserted_item in pda/New().